android/java/org/chromium/chrome/browser/crypto_wallet/fragments/TxDetailsFragment.java

```diff
@@ -34,6 +42,79 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
             @Nullable Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_transaction_details, container, false);
 
+        setupView(view);
+
         return view;
     }
+
+    @SuppressLint("ClickableViewAccessibility")
+    public void setupView(View view) {
+        if ((mTxInfo.txParams.length == 0 || mTxInfo.txArgs.length == 0)
+                && mTxInfo.txData.baseData.data.length == 0) {
+            return;
+        }
+        assert mTxInfo.txParams.length == mTxInfo.txArgs.length;
+        String functionType = getResources().getString(R.string.wallet_details_function_type_other);
+        switch (mTxInfo.txType) {
+            case TransactionType.ERC20_TRANSFER:
+                functionType = getResources().getString(
+                        R.string.wallet_details_function_type_erc20transfer);
+                break;
+            case TransactionType.ERC20_APPROVE:
+                functionType = getResources().getString(
+                        R.string.wallet_details_function_type_erc20approve);
+                break;
+            case TransactionType.ERC721_TRANSFER_FROM:
```

Comment:
FYI we just have another type added, ERC721_SAFE_TRANSFER_FROM.

Reply:
kk, I just added that type for the future. No Erc721 support on Android yet. So let's leave it for when the support is going to be implemented.